ethereum/contracts/DOTApp.sol

```diff
@@ -97,4 +108,23 @@ contract DOTApp is FeeSource, AccessControl {
                 _amount.encode256()
             );
     }
+
+    function upgrade(
```

Comment:
@wildmolasses 

should we consider moving all this duplicate code into something like a shared ChannelApp.sol contract that these can inherit from? maybe in a follow up PR though if so

Reply:
Sure, that sounds good.